templates/deployments/appname/app.jsonnet.tpl

```diff
@@ -24,6 +25,35 @@ local sharedLabels = {
 	reporting_team: '{{ stencil.Arg "reportingTeam" }}',
 };
 
+{{- if eq "canary" (stencil.Arg "deployment.strategy") }}
+local deploymentMetrics = [
```

Comment:
Should we make this an object so it's easier to override these? Not super important, but per object array overrides is a pain in jsonnet

Reply:
it's already an object '[analysis_template](https://github.com/getoutreach/stencil-golang/pull/192/files#diff-1af69013f6d48bcf54ab5b69c643cda18cb1f9cca59dc60a8c817ef9cf3d6840R177-R179)' which can be overridden as example below

analysis_template+: {
    spec+: {
      metrics+: [
        argo.AnalysisMetricDatadog('cpu-usage') {
          query:: 'moving_rollup(default_zero(100 * avg:kubernetes.cpu.usage.total{app:%(name)s,kube_namespace:%(namespace)s,image_tag:%(version)s}) / 1000000000 / avg:kubernetes.cpu.limits{app:%(name)s,kube_namespace:%(namespace)s,image_tag:%(version)s}, 60, "max")' % app,
          successCondition: 'default(result, 0) < 70',
          interval: '1m',
	 },
      ],
    },
  },